### DISCUSSION TITLE
Postgres database

### DISCUSSION THREAD

**Discussion by jithendra945** [cuebook/CueObserve#55] [Q&A] [answered]
I tried to give postgres as given in the document.
But it is on my local system.

When i run it, it is giving 500 internal server error, could not connect to postgres server.

I can connect to postgres psql on my system but this project is not connecting.

Anyone know the solution ?

**Comment by sachinkbansal** [comment, marked as answer]
@jithendra945 we'll check and get back

**Reply by jithendra945** [reply]
Ok. I would appreciate it.
Please make some videos on how to add connection and add dataset too.

I tried adding BigQuery as a connection. But in Datasets adding, im getting error.

**Reply by sachinkbansal** [reply]
@jithendra945 I am unable to reproduce BigQuery issue. I just created a new connection and added a dataset.

When are you getting the error? When you `Run SQL` or when you try to `Save Dataset`.
If it's the first, can you please share your SQL so that we can have a look.
If it's the latter, please share a screenshot of the dataset schema along with the SQL.

**Reply by jithendra945** [reply]
ok, i will recheck with BigQuery.

Have youu checked with postgres as mentioned in https://cueobserve.cuebook.ai/installation Use Postgres as the application database.

**Reply by sachinkbansal** [reply]
I have created an [issue](https://github.com/cuebook/CueObserve/issues/56) for the postgres. External postgres works but there's an issue connecting to a local postgres server

**Reply by jithendra945** [reply]
and Sir, can you provide some detail Instructions on running the project after cloning the repo.
I want to check the Exception while running query.
I don't know Docker, so I'm not getting how to run it

**Reply by vikrantcue** [reply]
Hi @jithendra945,

We have created an issue to add the documentation for development on local machine #57.

Meanwhile to get you started with development here is a little guide you can follow:

**1. Running the UI service:**
Modify the api service for local development
Go to ui/src/services/api.js 
And comment line number 8-9 and uncomment 4-5.
Then follow the following steps to start the react application:
```
cd ui
npm install 
npm start
```
This will start the UI service on port 3000.


**2. Running the backend service**
```
cd api

# Install virtualenv package if you don't have it installed already
pip install virtualenv 

# Create a new virtual environment
virtualenv -p python3 env

# Activate your newly created virtual environment
source env/bin/activate

# Install the dependencies for the project
pip install -r requirements.txt

# Modify .env.dev to have your local DB credentials and then source it to set all the environment variables
source .env.dev

# Migrate the database
python manage.py migrate

# Run django server
python manage.py runserver
```
This will start the django server on port 8000. 

Now go to http://localhost:3000 and see if your UI is able to connect to the backend service on port 8000. 

Let me know if you face any issues.

**Reply by jithendra945** [reply]
Thank you. Appreciate it.
